include/zephyr/net/lwm2m.h

```diff
@@ -1410,7 +1410,23 @@ void lwm2m_rd_client_update(void);
 char *lwm2m_path_log_buf(char *buf, struct lwm2m_obj_path *path);
 
 /** 
- * @brief LwM2M SEND operation to given path list
+ * @brief LwM2M SEND operation to given path list asynchronously with confirmation callback
+ *
+ * @param ctx LwM2M context
+ * @param path_list LwM2M Path string list
+ * @param path_list_size Length of path list. Max size is CONFIG_LWM2M_COMPOSITE_PATH_LIST_SIZE
+ * @param reply_cb Callback triggered with confirmation state.
+ * 
+ * @return 0 for success or negative in case of error.
+ *
+ */
+int lwm2m_engine_send_cb(struct lwm2m_ctx *ctx, char const *path_list[], uint8_t path_list_size,
+		     coap_reply_t *reply_cb);
+
+/** 
+ * @brief LwM2M SEND operation to given path list asynchronously without confirmation
+ *
+ * @deprecated Use lwm2m_engine_send_cb() instead, to check for confirmation request callbacks
```

Comment:
Use also `__deprecated` before the function so compiler can warn about it.

Reply:
Done